api/httpjson/RPCserver.go

```diff
@@ -64,6 +64,9 @@ func (s *RPCServer) write(w http.ResponseWriter, data []byte) {
 //this is the funciton that should be called in order to answer an rpc call
 //should be registered like "http.HandleFunc("/", httpjsonrpc.Handle)"
 func (s *RPCServer) Handle(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Access-Control-Allow-Headers", "Content-Type")
+	w.Header().Set("content-type", "application/json;charset=utf-8")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
I prefer move these lines to the read lock.

Reply:
Done.

Reply:
Updated and also removed unused function.